src/classes/RecordTypeService_TEST.cls

```diff
@@ -0,0 +1,146 @@
+        System.assertEquals(true, recordTypeServiceInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that a map of Record Type model by Id is returned by getModelByIdMapForSObjectType correctly
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getModelByIdMapForSObjectTypeTest() {
+        RecordTypeMapper.instance = new STUB_RecordTypeMapper();
+
+        Test.startTest();
+            Map<Id, RecordTypeModel> recordTypeModelById = RecordTypeService.getInstance().getModelByIdMapForSObjectType(Account.getSobjectType());
+        Test.stopTest();
+
+        System.assertEquals(true, recordTypeModelById.keySet().contains('000000000000000000'));
+        System.assertEquals('Household', recordTypeModelById.get('000000000000000000').name);
+        System.assertEquals('Household Account', recordTypeModelById.get('000000000000000000').developerName);
+        System.assertEquals('hed', recordTypeModelById.get('000000000000000000').namespace);
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that null is returned for valid id not in the map
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getRecordTypeModelFromMapValidIdNotInMapTest() {
+        RecordTypeMapper.instance = new STUB_RecordTypeMapper();
+
+        Test.startTest();
+            Map<Id, RecordTypeModel> recordTypeModelById = RecordTypeService.getInstance().getModelByIdMapForSObjectType(Account.getSobjectType());
+            System.assertEquals(null, RecordTypeService.getInstance().getRecordTypeModelFromMap('000000000000000001', recordTypeModelById));
+        Test.stopTest();
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that null is returned for invalid id
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getRecordTypeModelFromMapInvalidIdTest() {
+        RecordTypeMapper.instance = new STUB_RecordTypeMapper();
+
+        Test.startTest();
+            Map<Id, RecordTypeModel> recordTypeModelById = RecordTypeService.getInstance().getModelByIdMapForSObjectType(Account.getSobjectType());
+            System.assertEquals(null, RecordTypeService.getInstance().getRecordTypeModelFromMap('invalid', recordTypeModelById));
+        Test.stopTest();
```

Comment:
We should be careful about including only the specific method we are testing inside of our test blocks so we can isolate those metrics from any test setup or asserts.
```suggestion
        Test.startTest();
            Map<Id, RecordTypeModel> recordTypeModelById = RecordTypeService.getInstance().getModelByIdMapForSObjectType(Account.getSobjectType());
        Test.stopTest();
        
        System.assertEquals(null, RecordTypeService.getInstance().getRecordTypeModelFromMap('invalid', recordTypeModelById));
```